### PR TITLE
Store WriteConcern in WriteResult to allow reporting of acknowledgement

### PR DESCRIPTION
**Pull request by jmikola**
* https://jira.mongodb.org/browse/PHPC-168
 * https://jira.mongodb.org/browse/PHPC-170
 * https://jira.mongodb.org/browse/PHPC-177

### CODE REVIEW

File: php_phongo.c [hunk 1]
```diff
@@ -1051,21 +1060,18 @@ void php_phongo_read_preference_to_zval(zval *retval, mongoc_read_prefs_t *read_
...
 		add_assoc_string_ex(retval, ZEND_STRS("w"), wtag, 1);
+	} else if (mongoc_write_concern_get_wmajority(write_concern)) {
+		add_assoc_string_ex(retval, ZEND_STRS("w"), "majority", 1);
 	} else {
-		if (!mongoc_write_concern_get_wmajority(write_concern)) {
-			add_assoc_long_ex(retval, ZEND_STRS("w"), mongoc_write_concern_get_w(write_concern));
-		}
+		const int32_t w = mongoc_write_concern_get_w(write_concern);
+
+		add_assoc_long_ex(retval, ZEND_STRS("w"), w == MONGOC_WRITE_CONCERN_W_DEFAULT ? 1 : w);
```
Comment (bjori) [inline php_phongo.c]: Thats not true.
It may feel like w=1, but it isn't. It means server defaults, which can be 5, or majority, or even 0.

Comment (bjori) [inline php_phongo.c]: See http://docs.mongodb.org/manual/reference/command/replSetGetConfig/#replSetGetConfig.settings.getLastErrorDefaults

File: php_phongo.c [hunk 2]
```diff
@@ -1051,21 +1060,18 @@ void php_phongo_read_preference_to_zval(zval *retval, mongoc_read_prefs_t *read_
 
 void php_phongo_write_concern_to_zval(zval *retval, mongoc_write_concern_t *write_concern) /* {{{ */
 {
-	char                   *wtag;
+	const char *wtag = mongoc_write_concern_get_wtag(write_concern);
 
 	array_init_size(retval, 5);
 
-	if (mongoc_write_concern_get_w(write_concern) == MONGOC_WRITE_CONCERN_W_DEFAULT) {
-		return;
```
Comment (bjori) [inline php_phongo.c]: This was relatively intentional.
If w=MONGOC_WRITE_CONCERN_W_DEFAULT, and fsync and j aren't specifcally provided, then there is no writeConcern.
In pre-2.6 it means that getLastError will be called, with no arguments, and the server defaults applied.
In 2.6+ it means no writeConcern document provided with the write, and the server defaults applied

I'm not sure how mongoc deals with fsync and j defaults.. I guess it doesn't care if they are explicitly provided or not (e.g. overwrite server defaults if explicitly provided).......

**Comment by bjori** [comment]
Other then the comments above, lgtm

File: php_phongo.c (hunk 2 above)
Comment (jmikola) [inline php_phongo.c]: I changed this to omit `w` if it's equal to `MONGOC_WRITE_CONCERN_W_DEFAULT`. We'll still print `wtag`, `"majority"`, or the numeric value for all other cases.

I'd like to do the same for `journal` and `fsync`, but that depends on mongodb/mongo-c-driver#194 being merged. As for `wtimeout`, 0 is considered the empty value for that. Ultimately, I think we need support for `null` arguments to WriteConcern or to simply take an `$options` array and be done with it. One argument is that we're locking ourselves into having an argument for `fsync`, which we already know is deprecated.

**Pull request merged**